Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor.cs

```diff
@@ -49,36 +53,252 @@ public partial class HxMultiSelectInternal<TValue, TItem>
 	/// </summary>
 	[Parameter] public RenderFragment InputGroupEndTemplate { get; set; }
 
+	/// <summary>
+	/// Enables filtering capabilities.
+	/// </summary>
+	[Parameter] public bool AllowFiltering { get; set; }
+
+	[Parameter] public Func<TItem, string, bool> FilterPredicate { get; set; }
+
+	[Parameter] public bool ClearFilterOnHide { get; set; }
+
+	/// <summary>
+	/// This event is fired when a dropdown element has been made visible to the user.
+	/// </summary>
+	[Parameter] public EventCallback<string> OnShown { get; set; }
+
+	/// <summary>
+	/// This event is fired when a dropdown element has been hidden from the user.
+	/// </summary>
+	[Parameter] public EventCallback<string> OnHidden { get; set; }
+
+	/// <summary>
+	/// Template that defines what should be rendered in case of empty items.
+	/// </summary>
+	[Parameter] public RenderFragment EmptyTemplate { get; set; }
+
+	[Parameter] public bool AllowSelectAll { get; set; }
+
+	[Parameter] public EventCallback<bool> SelectAllChanged { get; set; }
+
+	[Parameter] public string SelectAllText { get; set; }
+
 	/// <summary>
 	/// Additional attributes to be splatted onto an underlying HTML element.
 	/// </summary>
 	[Parameter(CaptureUnmatchedValues = true)] public Dictionary<string, object> AdditionalAttributes { get; set; }
 
+	[Inject] private IJSRuntime JSRuntime { get; set; } = default!;
```

Comment:
Remove nullable-references notation, we do not use nullable-references feature in our code.